openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -133,6 +89,307 @@ def no_workspace_dir():
+
+        Args:
+            source (str): Path to source file.
+            colorspace (str): Colorspace of the source file.
+            color_management (dict): Maya Color management data from
+                `lib.get_color_management_preferences`
+            staging_dir (str): Output directory to write to.
+
+        Returns:
+            TextureResult: The resulting texture information.
+
+        """
+
+        maketx_path = get_oiio_tools_path("maketx")
+
+        if not maketx_path:
+            raise AssertionError(
+                "OIIO 'maketx' tool not found. Result: {}".format(maketx_path)
+            )
+
+        # Define .tx filepath in staging if source file is not .tx
+        fname, ext = os.path.splitext(os.path.basename(source))
+        if ext == ".tx":
+            # Do nothing if the source file is already a .tx file.
+            return TextureResult(
+                path=source,
+                file_hash=None,     # todo: unknown texture hash?
+                colorspace=colorspace,
+                transfer_mode=COPY
+            )
+
+        args = []
+        if color_management["enabled"]:
+            config_path = color_management["config"]
+            if not os.path.exists(config_path):
+                raise RuntimeError("OCIO config not found at: "
+                                   "{}".format(config_path))
+
+            render_colorspace = color_management["rendering_space"]
+
+            self.log.info("tx: converting colorspace {0} "
+                          "-> {1}".format(colorspace,
+                                          render_colorspace))
+            args.extend(["--colorconvert", colorspace, render_colorspace])
+            args.extend(["--colorconfig", config_path])
+
+        else:
+            # We can't rely on the colorspace attribute when not in color
+            # managed mode because the collected color space is the color space
+            # attribute of the file node which can be any string whatsoever
+            # but only appears disabled in Attribute Editor. We assume we're
+            # always converting to linear/Raw if the source file is assumed to
+            # be sRGB.
+            # TODO Without color management do we even know we can do
+            #      "colorconvert" and what config does that end up using since
+            #       colorconvert is a OCIO command line flag for maketx.
+            #       Also, Raw != linear?
+            render_colorspace = "linear"
+            if self._has_arnold():
+                img_info = image_info(source)
+                color_space = guess_colorspace(img_info)
+                if color_space.lower() == "sRGB":
+                    self.log.info("tx: converting sRGB -> linear")
+                    args.extend(["--colorconvert", "sRGB", "Raw"])
+                else:
+                    self.log.info("tx: texture's colorspace "
+                                  "is already linear")
+            else:
```

Comment:
if using `.tx` most likely arnold would already be included but I think my ordering would be something like this:
```
if ext.lower() in {".png", ".jpg", ".jpeg"}:
    input_colorspace = "sRGB"
elif self._has_arnold():
    img_info = image_info(source)
    input_colorspace = guess_colorspace(img_info)
else:
    self.log.warning(
        "tx: cannot guess the colorspace, color conversion won't be available!"
    )
    continue

if input_colorspace.lower() == "sRGB":
    self.log.info("tx: converting %s -> %s." % (input_colorspace, render_colorspace))
    args.extend(["--colorconvert", "sRGB", render_colorspace])
```
And I'd probably try store the `{".png", ".jpg", ".jpeg"}` set as some constant variable somewhere like `NONLINEAR_FILE_FORMATS`

Reply:
> if using .tx most likely arnold would already be included

`.tx` is also supported by other renderers (or anything supporting OpenImageIO I believe?). I know that e.g. V-Ray can render `.tx` files.

---

> ```
> if ext.lower() in {".png", ".jpg", ".jpeg"}:
>     input_colorspace = "sRGB"
> ```

I'm fine with adding this. @moonyuet @iLLiCiTiT thoughts? Can we assume any of those formats to ALWAYS be sRGB or do you predict any issues? Note that this logic is only used if Color Management is disabled in Maya.

Reply:
> > if using .tx most likely arnold would already be included
> 
> `.tx` is also supported by other renderers (or anything supporting OpenImageIO I believe?). I know that e.g. V-Ray can render `.tx` files.
> 
> > ```
> > if ext.lower() in {".png", ".jpg", ".jpeg"}:
> >     input_colorspace = "sRGB"
> > ```
> 
> I'm fine with adding this. @moonyuet @iLLiCiTiT thoughts? Can we assume any of those formats to ALWAYS be sRGB or do you predict any issues? Note that this logic is only used if Color Management is disabled in Maya.

I am not sure about png as someone claims that there is also 16-bit or 32-bit png despite the rare existence(http://www.libpng.org/pub/png/png-rgba32.html). But adding jpg and jpeg are definitely fine for me

Reply:
Thanks for these references. They are good references for #4713 but since I've been able to now avoid the color space guessing since conversion couldn't happen anyway with color management disabled I suppose it's not needed for this particular PR.